src/components/DeviceItem/index.js

```diff
@@ -69,13 +84,30 @@ class DeviceItem extends Component {
               type="text"
               value={curDescription}
               onChange={e => {
-                this.setState({ description: e.target.value })
+                this.setState({ curDescription: e.target.value })
               }}
             />
           ) : (
-            description
+            curDescription
           )}
         </ItemDescription>
+        <ItemPlaceSelector>
+          {editState ? (
+            <PlacesContext.Consumer>
```

Comment:
Since PlaceSelector will always need access to the PlacesContext.Consumer, we can move this complexity into the PlaceSelector component.